oneview_redfish_toolkit/multiple_oneview.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2018) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Python libs
+import logging
+
+# 3rd party libs
+from hpOneView.exceptions import HPOneViewException
+
+# Modules own libs
+from oneview_redfish_toolkit import authentication
+from oneview_redfish_toolkit import connection
+
+# Globals vars:
+#   globals()['map_resources_ov']
+
+
+NOT_FOUND_ERROR = ['RESOURCE_NOT_FOUND', 'ProfileNotFoundException',
+                   'DFRM_SAS_LOGICAL_JBOD_NOT_FOUND']
+
+
+def init_map_resources():
+    globals()['map_resources_ov'] = dict()
+
+
+def get_map_resources():
+    return globals()['map_resources_ov']
+
+
+def set_map_resources_entry(resource_id, ip_oneview):
+    get_map_resources()[resource_id] = ip_oneview
+
+
+def query_ov_client_by_resource(resource_id, resource, function,
+                                *args, **kwargs):
+    """Query resource on OneViews.
+
+        Query specific resource ID on multiple OneViews.
+        Look resource ID on cached map ResourceID->OneViewIP for query
+        on specific cached OneView IP.
+        If the resource ID is not cached yet it searchs on all OneViews.
+
+        Returns:
+            dict: OneView resource
+    """
+    # Get cached OneView IP by resource ID
+    ip_oneview = get_ov_ip_by_resource(resource_id)
+
+    # If resource is not cached yet search in all OneViews
+    if not ip_oneview:
+        return search_resource_multiple_ov(resource, function, resource_id,
+                                           *args, **kwargs)
+
+    # Get cached OneView's token
+    ov_token = authentication.get_oneview_token(ip_oneview)
+
+    ov_client = connection.get_oneview_client(ip_oneview, token=ov_token)
+
+    return execute_query_ov_client(ov_client, resource, function,
+                                   *args, **kwargs)
+
+
+def get_ov_ip_by_resource(resource_id):
+    """Get cached OneView's IP by resource ID"""
+    map_resources = get_map_resources()
+
+    if resource_id not in map_resources:
```

Comment:
You could simply execute `ip_oneview = map_resources.get(resource_id)` and return this value, right?! 
Instead of verifying if resource_id is in map_resources and to get the value of `ip_oneview`.

But, no problem keeping this code.

Reply:
Done.

Reply:
Done.